Automattic/wp-calypso
# Plans Grid: add monthly pricing option

### Changes proposed in this Pull Request

* Add the `<PlansIntervalToggle>` component to the Plans Grid UI
* Add various UI elements to the Plans Grid (both table and accordion variations) related to monthly plans
* Rewrite the Feature List component, in order to allow any feature to be showing the "requires annually plan" badge
* Hook the UI to the plans data-store, using the billing period selected by the user (monthly or annually) and update the plans accordion / table accordingly
* Updated Step by Step launch flow, so that plan info is displayed correctly in the final checkout step. When the monthly plan is selected, the cost of the premium domain is displayed in the final step too (plus minor style fixes)
* Update the refund window (14 or 7 days) in the plans step in Gutenboarding, based on the value picked by the user when interacting with the plans grid toggle

### Testing instructions

#### Setup

#### Test in Gutenboarding

- Pull this PR's branch on your local machine or test using the live link: https://calypso.live/new?branch=feat/plans-grid/monthly-prices
- If Run `yarn && yarn start` in the root folder of this repo
- Visit `calyspo.localhost:3000/new` and navigate to the Plan step
- Make sure that all A/Cs from #48760 are satisfied (except the follow up/related issues)

#### Test in Focused Launch Flow

- Prepare your sandbox:
  - Make sure you have an active ssh connection to your sandbox
  - Clear your sandbox and pull the latest version from trunk
  - Add an unlaunched site **created via `/start`** to your sandbox
- Pull this PR's branch on your local machine and run the project locally:
  - Run `yarn && yarn start` in the root folder
  - After the command above has finished compiling, in a separate terminal window run `cd apps/editing-toolkit && yarn dev --sync`
- Visit `calyspo.localhost:3000/page/UNLAUNCHED_SITE_CREATED_WITH_START.wordpress.com/home?flags=create/focused-launch-flow`
- Click on the "Launch" button to open Focused Launch flow
- Click on the "View all Plans" button to navigate to the Plan Details step
- Make sure that all A/Cs from #48760 are satisfied (except the follow up/related issues)


#### Test in Step By Step Launch Flow

- Prepare your sandbox:
  - Make sure you have an active ssh connection to your sandbox
  - Clear your sandbox and pull the latest version from trunk
  - Add an unlaunched site **created via `/new`** and **on a free plan** to your sandbox
- Pull this PR's branch on your local machine and run the project locally:
  - Run `yarn && yarn start` in the root folder
  - After the command above has finished compiling, in a separate terminal window run `cd apps/editing-toolkit && yarn dev --sync`
- Visit `calyspo.localhost:3000/page/UNLAUNCHED_SITE_CREATED_WITH_NEW .wordpress.com/home`
- Click on the "Launch" button to open Step By Step Launch flow
- Proceed to the plans step
- Make sure that all A/Cs from #48760 are satisfied (except the follow up/related issues)
- Make sure that checkout step looks correct (all of the listed features are correctly included in the selected plan, including a custom domain in case of annually-billed plans)

Fixes #48760
Fixes #49027
